test/util/removeUnnecessaryParens.test.js

```diff
@@ -22,7 +22,7 @@ describe('removeUnnecessaryParens', function () {
     ['(((-5)))', '-5'],
     ['((4+5)) + ((2^3))', '(4 + 5) + 2^3'],
     ['(2x^6 + -50 x^2) - (x^4)', '2x^6 - 50x^2 - x^4'],
-    ['(x+4) - (12 + x)', 'x + 4 - (12 + x)'],
+    ['(x+4) - (12 + x)', '(x + 4) - (12 + x)'],
```

Comment:
I would expect `x + 4 - (12 + x)` actually, you don't need parenthesis around x and 4

and when flattened you should get three children of +: `x`, `4`, and `-(12 + x)`